packages/orm/src/relations/OneToManyCollection.ts

```diff
@@ -199,7 +199,7 @@ export class OneToManyCollection<T extends Entity, U extends Entity>
     const current = await this.load({ withDeleted: true });
     current.forEach((other) => {
       const m2o = this.getOtherRelation(other);
-      if (maybeResolveReferenceToId(m2o.current()) === this.entity.id) {
+      if (maybeResolveReferenceToId(m2o.current({ withDeleted: true })) === this.entity.id) {
```

Comment:
thought: I wonder if an audit of use cases of `current` for other missing `withDeleted` clauses is warranted.

Reply:
Likely...fwiw I believe `current` is an internal method, so I wouldn't be against draconian methods like _requiring_ `withDeleted` just for good measure. (Dunno, that would get really ugly, but just talking out loud.)